lib/shopify_cli/theme/extension/syncer/extension_serve_job.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require "project_types/extension/loaders/project"
+require "project_types/extension/loaders/specification_handler"
+require "shopify_cli/partners_api"
+
+module ShopifyCLI
+  module Theme
+    module Extension
+      class Syncer
+        class ExtensionServeJob < ThreadPool::Job
+          POLL_FREQUENCY = 1 #second
+          PUSH_INTERVAL = 5 #seconds
+
+          RESPONSE_FIELD = %w(data extensionUpdateDraft)
+
+          def initialize(ctx, syncer:, extension:)
+            super(POLL_FREQUENCY)
+
+            @ctx = ctx
+            @extension = extension
+            @syncer = syncer
+
+            @mut = Mutex.new
+          end
+
+          def perform!
+            return if @syncer.pending_updates.empty? # if no updates
+            return if Time.now - @syncer.latest_sync < PUSH_INTERVAL
+
+            project = ::Extension::Loaders::Project.load(
+              context: @ctx,
+              directory: Dir.pwd,
+              api_key: nil,
+              api_secret: nil,
+              registration_id: nil,
+            )
+
+            specification_handler = ::Extension::Loaders::SpecificationHandler.load(project: project, context: @ctx)
+
+            input = {
+              api_key: project.app.api_key,
+              registration_id: project.registration_id,
+              config: JSON.generate(specification_handler.config(@ctx)),
+              extension_context: specification_handler.extension_context(@ctx)
+            }
+            response = ShopifyCLI::PartnersAPI.query(@ctx, "extension_update_draft", **input).dig(*RESPONSE_FIELD)
+            puts "RES: #{response}"
```

Comment:
I don't think we should retry when we face an error, because in the context of theme extensions we're uploading everything in a single request.

Thus, I believe it would make sense to just report the error with `@ctx.error`.

Also, when everything goes well, I might show a success message, following the same standard we have for themes:

<img width="452" alt="image" src="https://user-images.githubusercontent.com/1079279/183892123-b5ca41bd-e7af-4755-b684-a79613e7a74c.png">